### PR TITLE
Implement VersionFormatCheck Validator

### PR DESCRIPTION
**Pull request by yongyang**
https://projects.engineering.redhat.com/browse/PRODCI-34

### CODE REVIEW

File: harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersion.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
  */
 public enum ClassVersion
 {
-    JAVA_12( 46 ), JAVA_13( 47 ), JAVA_14( 48 ), JAVA_15( 49 ), JAVA_16( 50 ), JAVA_17( 51 ), JAVA_18( 52 );
+    JAVA_0(0), JAVA_12( 46 ), JAVA_13( 47 ), JAVA_14( 48 ), JAVA_15( 49 ), JAVA_16( 50 ), JAVA_17( 51 ), JAVA_18( 52 );
```
Comment (rnc) [inline harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersion.java]: What is java version 0 please?

File: harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersionInspector.java [hunk 2]
```diff
@@ -101,4 +97,43 @@ public static boolean checkJarVersion (String filename, ClassVersion version) th
...
+    /**
+     * Support version scope
+     * @param filename
+     * @param minVersion
+     * @param maxVersion
+     * @return
+     * @throws PVTException
+     */
+    public static boolean checkJarVersion (String filename, ClassVersion minVersion, ClassVersion maxVersion) throws PVTException
+    {
```
Comment (rnc) [inline harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersionInspector.java]: If this is for PRODCI-34 then that is version format check - nothing to do with version of the classes (which is jdkcompatiblecheck) ? Can this be removed and moved into a separate PR please

File: harness/src/main/java/org/jboss/pvt/harness/validators/VersionFormatCheckValidator.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        for (File jarFile : jarFiles) {
+            if (jarFile.isFile()) {
+
+                if (isFilter(pvtConfiguration, jarFile)) {
+                    logger.warn("Excluding Jar, ", jarFile);
+                    continue;
+                }
+
+                if(!pattern.matcher(jarFile.getName()).matches()) {
+                    failedJars.add(jarFile);
```
Comment (rnc) [inline harness/src/main/java/org/jboss/pvt/harness/validators/VersionFormatCheckValidator.java]: I don't think this will validate OSGi compatibility. Please use the code suggested in PRODCI-34 (contact me offline if you need help on this).

File: harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersionInspector.java (hunk 2 above)
Comment (ryanzhang) [inline harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersionInspector.java]: Agreed that it would be good practices to seperate different fix into separate ticket so we can have better track on things and easier for code review.

@yongyang, Can I suggest to try  to do that seperation as a best pratices?
Personally, I am ok to keep this as it was to move forward.

File: harness/src/main/java/org/jboss/pvt/harness/validators/VersionFormatCheckValidator.java (hunk 3 above)
Comment (ryanzhang) [inline harness/src/main/java/org/jboss/pvt/harness/validators/VersionFormatCheckValidator.java]: Thanks very much for pointing the PME code. 
Yes, according to https://docs.engineering.redhat.com/display/JP/Productization+Generic+Test+Case, we do need the OSGi version format checking.
@yongyang It will be ideal to have the implementation. 
We might not be able to complete in one PR but would be ideal to have it

**Comment by ryanzhang** [comment]
+1
Since the check is passed, and considering this PR have been on hold for long time, I think it's   better   to move forward.

It would be better to add the OSGi part later.

File: harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersion.java (hunk 1 above)
Comment (yongyang) [inline harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersion.java]: java 0 is for supporting version scope [0, 52]

File: harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersionInspector.java (hunk 2 above)
Comment (yongyang) [inline harness/src/main/java/org/jboss/pvt/harness/utils/ClassVersionInspector.java]: I have many files to commit, choosed wrong files for this PR, this file is for jdkcompatible validator.
